src/main/java/bio/terra/resourcemanagement/service/google/GoogleResourceService.java

```diff
@@ -271,6 +283,32 @@ private void enableServices(GoogleProjectResource projectResource) {
         }
     }
 
+    public void enableIamPermissions(GoogleProjectResource projectResource) {
+        Map<String, List<String>> userPermissions = projectResource.getUserPermissions();
+        GetIamPolicyRequest getIamPolicyRequest = new GetIamPolicyRequest();
+
+        try {
+            CloudResourceManager resourceManager = cloudResourceManager();
+            Policy getIamPolicy = resourceManager.projects()
+                .getIamPolicy(projectResource.getGoogleProjectId(), getIamPolicyRequest).execute();
+            List<Binding> bindingsList = getIamPolicy.getBindings();
+
+            for (Map.Entry<String, List<String>> entry : userPermissions.entrySet()) {
+                Binding binding = new Binding()
+                    .setRole(entry.getKey())
+                    .setMembers(entry.getValue());
+                bindingsList.add(binding);
+            }
+
+            Policy policy = new Policy().setBindings(bindingsList);
```

Comment:
Instead of creating a new policy it is probably better to reuse the `getIamPolicy` that we fetched above. This would prevent us from overriding properties other than the bindings.

Reply:
Suggestion: instead of storing a map in userPermissions, why not create the Binding object when the GoogleProjectRequest is created. The userPermission map is never mutated; (in fact, a tiny code safety improvement would be to make it unmodifiable after construction of the Request).